### PR TITLE
Demographics Endpoint Bearer Token Validation

### PR DESCRIPTION
**Pull request by jorg3lopez**
# Demographics Endpoint Bearer Token Validation

Demographics endpoint will enforce a bearer token for all incoming requests.

## Issue

#148 

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [ ] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 1]
```diff
@@ -65,6 +67,18 @@ public String openApiSpecification() {
 
     DomainResponse handleOrder(DomainRequest request) {
 
+        // Call the DemographicsRequestValidator class
+        DemographicsRequestValidator validator = new DemographicsRequestValidator(request);
+
+        // Validate token
+        try {
+            if (!validator.isValidToken()) {
```
Comment (jorg3lopez) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:75]: demographics unit test fails here, due to logger being null.

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/jjwt/JjwtEngine.java [hunk 2]
```diff
@@ -113,6 +114,29 @@ public void validateToken(String jwt, String encodedKey)
...
+        try {
+            Jws<Claims> claimsJws =
+                    Jwts.parserBuilder()
+                            .setSigningKey(readPrivateKey(privateKey))
+                            .build()
+                            .parseClaimsJws(jwt);
+
+            return LocalDateTime.now()
+                    .isBefore(
+                            claimsJws
```
Comment (jorg3lopez) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/external/jjwt/JjwtEngine.java:129]: Another option here is to call the getExpirationDate() and passed the jwt param. But we might get a warning that we are not using the claimsJws object.

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java (hunk 1 above)
Comment (halprin) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java]: This has been resolved in the follow-on commits.

File: e2e/src/test/groovy/gov/hhs/cdc/trustedintermediary/e2e/AuthTest.groovy [hunk 3]
```diff
@@ -31,21 +28,21 @@ class AuthTest extends Specification {
...
     }
 
     def "a 401 response is returned when poorly formatted request"() {
         given:
-        def invalidRequest= "invalid-request"
+        def invalidRequest = "invalid-request"
 
         when:
-        def response = Client.post(authEndpointPath, invalidRequest, ContentType.APPLICATION_FORM_URLENCODED)
+        def response = Client.post(authEndpointPath, invalidRequest, ContentType.APPLICATION_FORM_URLENCODED, Map.of())
```
Comment (basiliskus) [inline e2e/src/test/groovy/gov/hhs/cdc/trustedintermediary/e2e/AuthTest.groovy:45]: Could be written using the new method like this:
`def response = AuthClient.loginRaw(invalidRequest, "asdf")`

**Review by basiliskus — COMMENTED** [review]
Great work! I just left one comment

File: e2e/src/test/groovy/gov/hhs/cdc/trustedintermediary/e2e/AuthTest.groovy (hunk 3 above)
Comment (jorg3lopez) [reply e2e/src/test/groovy/gov/hhs/cdc/trustedintermediary/e2e/AuthTest.groovy]: I changed it to AuthClient.

**Review by basiliskus — APPROVED** [review]

**Pull request merged**